### ISSUE TITLE
Encountered invalid operation on entities with non-matching bit-widths of BitWidth(64) and BitWidth(32).

### ISSUE THREAD

**Issue by Nwqda** [fkie-cad/cwe_checker#262]
Hello,

I have the following error while running cwe_checker on a particular binary only:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: UnmatchingBitwidth(BitWidth(64), BitWidth(32)), message: "Encountered invalid operation on entities with non-matching bit-widths of BitWidth(64) and BitWidth(32).", annotation: None }', src/cwe_checker_lib/src/intermediate_representation/bitvector.rs:245:57
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
``` 

Binary type:
```
ELF 32-bit LSB executable, ARM, EABI5 version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 3.2.0, stripped
```
Thank you.

**Comment by Enkelmann** [comment]
This is a bug, where somewhere in the code the bit-size of a value is incorrectly tracked. Can you share the binary in question? If not, could you run it with `RUST_BACKTRACE=1` (as the error message suggests) and share its output?

**Comment by Nwqda** [comment]
Hi @Enkelmann,
Thanks for your answer.

As you recommended I've set RUST_BACKTRACE to 1, here is the output:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: UnmatchingBitwidth(BitWidth(64), BitWidth(32)), message: "Encountered invalid operation on entities with non-matching bit-widths of BitWidth(64) and BitWidth(32).", annotation: None }', src/cwe_checker_lib/src/intermediate_representation/bitvector.rs:245:57
stack backtrace:
   0: rust_begin_unwind
             at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:517:5
   1: core::panicking::panic_fmt
             at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/panicking.rs:101:14
   2: core::result::unwrap_failed
             at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/result.rs:1617:5
   3: <apint::apint::ApInt as cwe_checker_lib::intermediate_representation::bitvector::BitvectorExtended>::signed_sub_overflow_checked
   4: cwe_checker_lib::abstract_domain::interval::simple_interval::Interval::sub
   5: cwe_checker_lib::abstract_domain::interval::bin_ops::<impl cwe_checker_lib::abstract_domain::interval::IntervalDomain>::sub
   6: cwe_checker_lib::abstract_domain::data::arithmetics::<impl cwe_checker_lib::abstract_domain::RegisterDomain for cwe_checker_lib::abstract_domain::data::DataDomain<T>>::bin_op
   7: cwe_checker_lib::analysis::pointer_inference::state::access_handling::<impl cwe_checker_lib::analysis::pointer_inference::state::State>::eval
   8: cwe_checker_lib::analysis::pointer_inference::state::access_handling::<impl cwe_checker_lib::analysis::pointer_inference::state::State>::eval
   9: cwe_checker_lib::analysis::pointer_inference::context::trait_impls::<impl cwe_checker_lib::analysis::forward_interprocedural_fixpoint::Context for cwe_checker_lib::analysis::pointer_inference::context::Context>::update_def
  10: <cwe_checker_lib::analysis::forward_interprocedural_fixpoint::GeneralizedContext<T> as cwe_checker_lib::analysis::fixpoint::Context>::update_edge
  11: cwe_checker_lib::analysis::fixpoint::Computation<T>::compute_with_max_steps
  12: cwe_checker_lib::analysis::pointer_inference::PointerInference::compute_with_speculative_entry_points
  13: cwe_checker_lib::analysis::pointer_inference::run
  14: cwe_checker_lib::AnalysisResults::compute_pointer_inference
  15: cwe_checker::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
If you want to debug it using my binary here is the link:
[cad.zip](https://github.com/fkie-cad/cwe_checker/files/7660662/cad.zip)

**Comment by Enkelmann** [comment]
Thank you for the information! I think I have found the bug and I am currently working on a fix.

**Comment by Enkelmann** [comment]
The fix (see PR #263) is now merged, closing the issue. Thank you again for reporting the bug!
